apps/http-server/packages/generic/src/storage/storage.ts

```diff
@@ -1,20 +1,51 @@
 import { Readable } from 'stream'
-import { CTX, CTXPost } from '../lib'
+import { CTXPost } from '../lib'
 
 export abstract class Storage {
 	abstract init(): Promise<void>
 	abstract getInfo(): string
-	abstract listPackages(ctx: CTX): Promise<true | BadResponse>
-	abstract headPackage(path: string, ctx: CTX): Promise<true | BadResponse>
-	abstract getPackage(path: string, ctx: CTX): Promise<true | BadResponse>
+	abstract listPackages(): Promise<
+		| {
+				sidecar: Sidecar
+				body: {
+					packages: PackageInfo[]
+				}
+		  }
+		| BadResponse
+	>
+	abstract headPackage(path: string): Promise<{ sidecar: Sidecar } | BadResponse>
+	abstract getPackage(path: string): Promise<{ sidecar: Sidecar; body: any } | BadResponse>
 	abstract postPackage(
 		path: string,
 		ctx: CTXPost,
 		fileStreamOrText: string | Readable | undefined
-	): Promise<true | BadResponse>
-	abstract deletePackage(path: string, ctx: CTXPost): Promise<true | BadResponse>
+	): Promise<{ sidecar: Sidecar; body: any } | BadResponse>
+	abstract deletePackage(path: string): Promise<{ sidecar: Sidecar; body: any } | BadResponse>
 }
 export interface BadResponse {
 	code: number
 	reason: string
 }
+export function isBadResponse(v: unknown): v is BadResponse {
+	return (
+		typeof v === 'object' &&
+		typeof (v as BadResponse).code === 'number' &&
+		typeof (v as BadResponse).reason === 'string'
+	)
+}
+
+export type PackageInfo = {
+	path: string
+	size: string
+	modified: string
+}
+
+export interface Sidecar {
+	statusCode: number
+	type?: string
+	length?: number
+	lastModified?: Date
+	headers: {
+		[key: string]: string
+	}
+}
```

Comment:
Because this is called "Sidecar" for a very long time I thought that there is now a "sidecar-file" somewhere that this represents. Took me a while to understand that this is just meta-information for the response body. Maybe a `ResponseMeta` name would be more fitting? I'm also wondering if making the `headers` optional would make stuff cleaner overall, since most users of the Sidecar interface just put an empty object there, and all the other fields apart from `statusCode` are optional anyway?

Reply:
Agreed on all points! Fixed now.